core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -252,12 +252,12 @@ protected void setArtificialWayTags(PointList pointList, ReaderWay way, double d
         // special handling for countries: since they are built-in with GraphHopper they are always fed to the EncodingManager
         Country country = Country.MISSING;
         for (CustomArea customArea : customAreas) {
-            Object countryCode = customArea.getProperties().get("ISO3166-1:alpha3");
-            if (countryCode == null)
+            Object id = customArea.getProperties().get(Country.JSON_ID);
```

Comment:
I think this is problematic, because it means that we consider every custom area with a property key `Country.JSON_ID` to be a country. And this likely happens, because the value of `Country.JSON_ID` is just "id". So far we identified countries among the custom areas by the existence of the (rather special) `"ISO3166-1:alpha3"` property.

Is it possible to rename the `id` property in the new country file?

Reply:
.. or alternatively set the alpha2/twoLetter property in `GHUtility#readCountries` and use this here. Currently we set the `"name"` property there (but do not use it) (?)

Reply:
Yes, thanks. See my comments inline. And it is certainly better to use a distinguished string like `ISO3166-1:alpha3` rather than something generic as `id` to identify the countries among all the custom areas .